azure-kusto-ingest/test/e2eTest.js

```diff
@@ -0,0 +1,170 @@
+            for (let item of testItems) {
+                await new Promise((resolve) => {
+                    ingestClient.ingestFromFile(item.path, item.ingestionProperties, async (err, results) => {
+                        if (err) assert.fail(`Failed to ingest ${item.description}`);
+                        resolve();
+                    });
+                });
+                await assertRowsCount(item);
+            }
+        });
+
+        it('ingestFromStream', async function () {
+            for (let item of testItems) {
+                let stream = fs.createReadStream(item.path);
+                if (item.path.endsWith('gz')) {
+                    stream = new StreamDescriptor(stream, null, CompressionType.GZIP);
+                }
+                await new Promise((resolve) => {
+                    ingestClient.ingestFromStream(stream, item.ingestionProperties, async (err, results) => {
+                        if (err) assert.fail(`Failed to ingest ${item.description}`);
+                        resolve();
+                    });
+                });
+                await assertRowsCount(item);
+            }
+        });
+    });
+
+    describe('StreamingIngestClient', function () {
+        it('ingestFromFile', async function () {
+            for (let item of testItems.filter(item => item.testOnstreamingIngestion)) {
+                await new Promise((resolve) => {
+                    streamingIngestClient.ingestFromFile(item.path, item.ingestionProperties, async (err, results) => {
+                        if (err) assert.fail(`Failed to ingest ${item.description}`);
+                        resolve();
+                    });
+                });
+                await assertRowsCount(item);
+            }
+        });
+
+        it('ingestFromStream', async function () {
+            for (let item of testItems.filter(item => item.testOnstreamingIngestion)) {
+                let stream = fs.createReadStream(item.path);
+                if (item.path.endsWith('gz')) {
+                    stream = new StreamDescriptor(stream, null, CompressionType.GZIP);
+                }
+                await new Promise((resolve) => {
+                    streamingIngestClient.ingestFromStream(stream, item.ingestionProperties, async (err, results) => {
+                        if (err) assert.fail(`Failed to ingest ${item.description}`);
+                        resolve();
+                    });
+                });
+                await assertRowsCount(item);
+            }
+        });
+    });
+});
+
+function sleep(ms) {
+    return new Promise((resolve) => { setTimeout(resolve, ms); });
+}
+
+function getTestResourcePath(name) {
+    return path.relative(process.cwd(), `azure-kusto-ingest/test/data/${name}`);
+}
+
+async function assertRowsCount(testItem) {
+    var count = 0;
+    var expected = testItem.rows;
+    for (var i = 0; i < 10; i++) {
+        await sleep(1000)
```

Comment:
Would 10 seconds suffice (we don't want tests to fail because some temporary delays)?

Reply:
It was sufficient for me, anyway I'll increase it